modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/api.ts

```diff
@@ -72,9 +72,23 @@ export async function getFields(fdsView: FDSViewType) {
 			return;
 		}
 
+		// Filter <field_name>_i18n properties from the UI
+		// We only need the <field_name> property for the field selection
+		// and hide the _i18n that shows the internals
+
+		const localizablePropertySuffix = '_i18n';
+
+		if (propertyValue.includes(localizablePropertySuffix)) {
+			return;
+		}
+
+		if (propertyValue.additionalProperties) {
+			return;
+		}
+
 		const type = propertyValue.type;
 
-		if (type === 'object' || type === 'array') {
```

Comment:
The `type === 'object'` check was preventing longText and richText fields to appear as selectable in the field list.

Reply:
we probably need to filter out most objects, except those that

1. Have a particular type, known by inspecting schema field `additionalProperties`
2. Have a name ending with `_i18n`

Supporting long texts sounds interesting. For rich texts we need to see if there's an appropriate renderer as, iinm, they're mostly html

So, it's likely we still need to check for  `type == object` but be a more lenient for specific cases

Reply:
Yeah, this is a problem, and it's not related to localization. Filtering out `type === 'object'` works well to catch almost all meta properties, besides `x-schema-name`, which we specifically handle. But, it also filters out LongText and RichText fields, which we don't want. All in all, it's not a great criteria for filtering.

Looking at `openapi.json`, I notices there is `extensions : {x-parent-map: "properties"}` property, that exists only on real fields. We can apply that criteria for Object resources. For other headless resources, we can keep the existing logic of filtering out `type === 'object'`. At least for now.

Ideally, we would have some way of explicitly distinguishing them. In OpenAPI convention, meta properties start with `x-`, which we follow only partially. It would be great if he had `x-actions`, `x-audit-events`, `x-taxonomy-category-ids` etc. Alternatively, "real" fields should have some explicit attribute, like `field: true`.